### PR TITLE
Write app to SSM in app init

### LINKED ISSUE #94: Implement app init
The App commands are not currently super useful, since their main behavior is taken care of by `init`, but once we start implementing `deploy` and `show`, it'll be useful to already have the namespace defined.

This task is for creating an `app init` command that:

1. Checks to see if we're in a WS
2. Prompts the user for the application *type* they want to build
3. Prompts the user for the application Dockerfile (right now - in the future we'll let them select creating a directory with one of our templated directories)
4. Writes the manifest to the workspace root
5. Writes the application into SSM

This task can be broken down into 2 parts:

1. Adding the Application to SSM (we currently only store projects and envs)
2. Wiring up the init command based on the current `ecs init` command

### PR DESCRIPTION
**Pull request by kohidave**
This change writes to SSM in the app init command. It
also checks to see that the app hasn't already been created
when running app init.

Fixes #94 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/cli/app_init.go [hunk 1]
```diff
@@ -103,6 +111,14 @@ func (opts *InitAppOpts) Execute() error {
 		return fmt.Errorf("relative path of manifest file: %w", err)
 	}
 	opts.manifestPath = relPath
```
Comment (efekarakus) [inline internal/pkg/cli/app_init.go]: nit: should we move all this manifest logic into a method? to make `Execute` easier to digest?
```
1. ensureNoExistingApp
2. createManifest
3. saveApp
🎉 success
```

File: internal/pkg/cli/app_init.go [hunk 2]
```diff
@@ -82,6 +85,11 @@ func (opts *InitAppOpts) Validate() error {
 
 // Execute writes the application's manifest file and stores the application in SSM.
 func (opts *InitAppOpts) Execute() error {
+	projectName := viper.GetString(projectFlag)
```
Comment (sonofachamp) [inline internal/pkg/cli/app_init.go]: Is it possible that `projectName` is empty/invalid here?

Comment (kohidave) [reply internal/pkg/cli/app_init.go]: In the Validate function we make sure it's not empty or invalid

File: internal/pkg/cli/app_init.go [hunk 3]
```diff
@@ -202,6 +218,21 @@ func (opts *InitAppOpts) listDockerfiles() ([]string, error) {
...
+	var existsErr *store.ErrNoSuchApplication
+	if errors.As(err, &existsErr) {
+		return nil
+	}
+	// If there's no error, that means we were able to fetch an existing app
+	if err == nil {
+		return fmt.Errorf("application %s already exists under project %s", appName, projectName)
+	}
+	// Otherwise, there was an error calling the store
+	return err
```
[the remark this answers is not in the document]
Comment (kohidave) [reply internal/pkg/cli/app_init.go]: Yummy.

File: internal/pkg/cli/app_init.go (hunk 1 above)
Comment (kohidave) [reply internal/pkg/cli/app_init.go]: Delightful!

**Pull request merged**